js/data/bucket/line_bucket.js

```diff
@@ -159,16 +159,18 @@ class LineBucket extends Bucket {
 
             // Determine the normal of the join extrusion. It is the angle bisector
             // of the segments between the previous line and the next line.
-            let joinNormal = prevNormal.add(nextNormal)._unit();
+            let joinNormal = prevNormal.add(nextNormal);
 
-            if (isNaN(joinNormal.x) && isNaN(joinNormal.y) && join === 'miter') {
+            if (joinNormal.x === 0 && joinNormal.y === 0 && join === 'miter') {
```

Comment:
this shouldn't include `join === 'miter'`; the division by zero happens in all cases. For non-miter joins, the `else` branch is still executed, leading to NaN values.